### PR TITLE
Update zboss libraries to v3.5.2.0

### PR DESCRIPTION
**Pull request by tomchy**
* Providing the two sets of libraries allowed us to add experimental features with the production version intact.
After the NCS v1.5.1 release, there will be only a single library to reduce the complexity of the Zigbee stack configuration.
* Update ZBOSS libraries to v3.5.2.0

### CODE REVIEW

**Review by greg-fer — CHANGES_REQUESTED** [review]
Zigbee memory requirements page also needs to be updated, as it contains tables for development and production libraries.

File: doc/nrf/ug_zigbee_configuring.rst
Comment (tomchy) [reply doc/nrf/ug_zigbee_configuring.rst:22]: Fixed.

File: samples/zigbee/ncp/README.rst
Comment (tomchy) [reply samples/zigbee/ncp/README.rst:52]: Fixed

File: subsys/zigbee/Kconfig
Comment (greg-fer) [inline subsys/zigbee/Kconfig:196]: Do not use `please`.

File: subsys/zigbee/Kconfig
Comment (greg-fer) [inline subsys/zigbee/Kconfig:194]:
```suggestion
	  For example, to enable traces related to OTA DFU, set this option to 0x00080100.
```

File: subsys/zigbee/Kconfig
Comment (greg-fer) [inline subsys/zigbee/Kconfig:164]:
```suggestion
	  Available modules:
```

File: subsys/zigbee/Kconfig
Comment (greg-fer) [inline subsys/zigbee/Kconfig:162]:
```suggestion
	  The mask value is expected to be a bitwise OR of values assigned to selected modules.
```

File: subsys/zigbee/Kconfig
Comment (greg-fer) [inline subsys/zigbee/Kconfig:173]: Shouldn't this be FOTA?
`ZCL: Firmware over-the-air upgrade`?

File: subsys/zigbee/Kconfig
Comment (greg-fer) [inline subsys/zigbee/Kconfig:176]: `MAC lower layer`?

File: subsys/zigbee/Kconfig
Comment (greg-fer) [inline subsys/zigbee/Kconfig:188]: `Application support layer`?

**Review by greg-fer — APPROVED** [review]
Thanks for addressing my comments.
Approving with some additional comments to Kconfig changes.

File: doc/nrf/ug_zigbee_configuring.rst
Comment (greg-fer) [inline doc/nrf/ug_zigbee_configuring.rst:24]: Will this ZBOSS Zigbee stack library be enabled by default after these changes are merged?
If yes, let's keep this line: ``This library is enabled by default with the :option:`name_of_option` Kconfig option.``

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: subsys/zigbee/Kconfig
Comment (tomchy) [reply subsys/zigbee/Kconfig:196]: Fixed.

File: subsys/zigbee/Kconfig
Comment (tomchy) [reply subsys/zigbee/Kconfig:194]: Fixed.

File: subsys/zigbee/Kconfig
Comment (tomchy) [reply subsys/zigbee/Kconfig:164]: Fixed.

File: subsys/zigbee/Kconfig
Comment (tomchy) [reply subsys/zigbee/Kconfig:162]: Fixed.

File: subsys/zigbee/Kconfig
Comment (tomchy) [reply subsys/zigbee/Kconfig:173]: I agree. It is better to keep it consistent and change to FOTA.

File: subsys/zigbee/Kconfig
Comment (tomchy) [reply subsys/zigbee/Kconfig:188]: Works for me

**Review by greg-fer — APPROVED** [review]

**Review by maciekfabia — APPROVED** [review]
LGTM

**Review by rlubos — APPROVED** [review]

**Pull request merged**
